### DISCUSSION TITLE
power consumption model and estimation

### DISCUSSION THREAD

**Discussion by ParhamMohamadi** [ctu-mrs/mrs_uav_system#159] [Q&A] [answered]
I was wondering to use ctu-mrs as the framework of my work. I could not find any power measurements topics or services in the [control_manager.launch](https://github.com/ctu-mrs/mrs_uav_managers/blob/master/launch/control_manager.launch) . I wonder if there is any further information about the battery usage and power consumption in mrs_uav_system.

thank you.

**Comment by klaxalk** [comment, marked as answer]
Hey, the particularities of power consumption are managed by the _Mavros_ node, which handles communication with the Pixhawk flight controller. You can get current and voltage measurements if you are running the Pixhawk's power board. However, the power board's sensors need to be calibrated first. Otherwise, the readings are going to be very skewed. If you do that, subscribe to the `/uav*/mavros/battery` topic (http://docs.ros.org/en/hydro/api/mavros/html/msg/BatteryStatus.html) to get the readings.
